http/1.6/modules/ROOT/pages/http-listener-configuration.adoc

```diff
@@ -0,0 +1,504 @@
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure the database connection.
+. In the *Insert* operation configuration screen, set *SQL Query Text* to `INSERT INTO logo (accountId,logoName)`.
+. Drag an *On Error Propagate* component in the *Error handling* section of the flow.
+. Set *Type* to `DB:CONNECTIVITY`.
+. Drag a *Set Variable* component into the *On Error Propagate* component.
+. Set *Name* to `errorCode`.
+. Set *Value* to `504`.
+
+=== XML for Sending HTTP Responses Mule App Example
+
+Paste this code into your Studio XML editor to quickly load the flow for this example into your Mule app:
+
+[source,xml,linenums]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+
+<mule xmlns:db="http://www.mulesoft.org/schema/mule/db" xmlns:os="http://www.mulesoft.org/schema/mule/os"
+	xmlns:http="http://www.mulesoft.org/schema/mule/http"
+	xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+http://www.mulesoft.org/schema/mule/os http://www.mulesoft.org/schema/mule/os/current/mule-os.xsd
+http://www.mulesoft.org/schema/mule/db http://www.mulesoft.org/schema/mule/db/current/mule-db.xsd">
+	<http:listener-config name="HTTP_Listener_config" >
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<db:config name="Database_Config" >
+		<db:derby-connection />
+	</db:config>
+	<flow name="httplistenerresponseFlow"  >
+		<http:listener config-ref="HTTP_Listener_config" path="/account/{accountId}/logo">
+			<http:response statusCode="201" reasonPhrase="Created">
+				<http:body ><![CDATA[output text/plain --- vars.logoName ++ ' has been stored as a ' ++ vars.accountId ++ ' logo.']]></http:body>
+			</http:response>
+			<http:error-response statusCode="#[vars.errorCode default 500]" >
+				<http:body ><![CDATA[vars.logoName ++ ' could not be stored.']]></http:body>
+				<http:headers ><![CDATA[#[output application/java
+        ---
+        {
+          "X-Time" : "50s" // 4
+        }]]]></http:headers>
+			</http:error-response>
+		</http:listener>
+		<set-variable value="payload.parts.name.content" variableName="logoName"/>
+		<set-variable value="attributes.uriParams.accountId"  variableName="accountId"/>
+		<db:insert doc:name="Insert" config-ref="Database_Config">
+			<db:sql ><![CDATA[INSERT INTO logo (accountId,logoName)]]></db:sql>
+		</db:insert>
+		<error-handler >
+			<on-error-propagate enableNotifications="true" logException="true" doc:name="On Error Propagate"  type="DB:CONNECTIVITY">
+				<set-variable value="504" variableName="errorCode"/>
+			</on-error-propagate>
+		</error-handler>
+	</flow>
+</mule>
+----
+
```

Comment:
To test it you need a table named logo with 2 columns: (accountId,logoName).
You can use mulesoft to create that table:
```
<db:execute-script config-ref="Database_Config">
  <db:sql ><![CDATA[CREATE TABLE logo (
	accountId VARCHAR(255),
	logoName VARCHAR(255),
	PRIMARY KEY (accountId)
)]]></db:sql>
</db:execute-script>
```
or do it some other way, but the table has to exist in order to the have the example working.

Once the table has been created and the app is started, you can use this curl to test it (I'm including my output so you can extend the doc):
```
curl -v -F 'data=@/path/to/MuleSoft_logo.svg' -F name=mulesoftlogo http://localhost:8081/account/muley/logo
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8081 (#0)
> POST /account/muley/logo HTTP/1.1
> Host: localhost:8081
> User-Agent: curl/7.64.1
> Accept: */*
> Content-Length: 5313
> Content-Type: multipart/form-data; boundary=------------------------d96e85d23101a1c3
> Expect: 100-continue
>
< HTTP/1.1 100 Continue
* We are completely uploaded and fine
< HTTP/1.1 201 Created
< Content-Type: text/plain; charset=UTF-8
< Content-Length: 45
< Date: Tue, 18 Jan 2022 20:30:20 GMT
<
* Connection #0 to host localhost left intact
mulesoftlogo has been stored as a muley logo.* Closing connection 0
```
Note that the complete payload has been hidden, because the content length is too big to be human readable.
The svg used in the example curl is irrelevant (in fact, it doesn't even need to be an image)